src/runtime_src/core/edge/skd/xrt_skd.h

```diff
@@ -83,37 +90,43 @@ class skd
   void
   fini();
 
+  void set_signal(int sig);
   void report_ready();
   void report_crash();
   void report_fini();
 
  private:
-    xclDeviceHandle parent_devHdl;
-    xclDeviceHandle devHdl;
-    xrtDeviceHandle xrtdHdl;
-    char sk_path[XRT_MAX_PATH_LENGTH];
-    uint32_t cu_idx;
-    char sk_name[PS_KERNEL_NAME_LENGTH];
-    pscontext* xrtHandle = NULL;
-    int sk_bo;
-    int sk_meta_bo;
-    unsigned char xclbin_uuid[16];
-
-    void* sk_handle;
-    void* kernel;
-    std::vector<xrt_core::xclbin::kernel_argument> args;
-    int cmd_boh = -1;
-    uint32_t *args_from_host;
-    ffi_type** ffi_args;
-    ffi_cif cif;
-    bool pass_xrtHandles = false;
-    int num_args;
-    int return_offset;
+    xclDeviceHandle m_parent_devhdl = 0;
+    xclDeviceHandle m_devhdl = 0;
+    xrtDeviceHandle m_xrtdhdl = 0;
+    uuid m_xclbin_uuid;
+    const std::filesystem::path m_sk_path;
+    uint32_t m_cu_idx = 0;
+    std::string m_sk_name = "";
+    pscontext* m_xrtHandle = nullptr;
+    int m_sk_bo = 0;
+    int m_sk_meta_bo = 0;
+    int m_parent_bo_handle = 0;
+    uint64_t m_mem_start_paddr = 0;
+    uint64_t m_mem_size = 0;
+    void* m_mem_start_vaddr = nullptr;
+    int signal = 0;
+    
+    void* m_sk_handle = nullptr;
+    void* m_kernel = nullptr;
+    std::vector<xrt_core::xclbin::kernel_argument> m_kernel_args;
+    int m_cmd_boh = -1;
+    uint32_t *m_args_from_host = nullptr;
+    std::vector<ffi_type*> m_ffi_args;
+    ffi_cif m_cif = {};
+    bool m_pass_xrtHandles = false;
+    int m_return_offset = 1;
     
-    int waitNextCmd();
-    int createSoftKernelFile(xclDeviceHandle handle, int bohdl);
-    int deleteSoftKernelFile();
-    int createSoftKernel(int *boh);
+    int wait_next_cmd();
+    int create_softkernelfile(xclDeviceHandle handle, int bohdl);
+    int delete_softkernelfile();
+    int create_softkernel(int *boh);
+    int get_return_offset(std::vector<xrt_core::xclbin::kernel_argument> args);
```

Comment:
Please review all the methods and update which parameters should be const or const reference.
Also review which methods are class constant (const after the method name)